app/bundles/LeadBundle/Form/FieldAliasToFqcnMap.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * @copyright   2019 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Form;
+
+use Mautic\CoreBundle\Form\Type\BooleanType;
+use Mautic\CoreBundle\Form\Type\CountryType;
+use Mautic\CoreBundle\Form\Type\LocaleType;
+use Mautic\CoreBundle\Form\Type\LookupType;
+use Mautic\CoreBundle\Form\Type\MultiselectType;
+use Mautic\CoreBundle\Form\Type\RegionType;
+use Mautic\CoreBundle\Form\Type\SelectType;
+use Mautic\CoreBundle\Form\Type\TelType;
+use Mautic\CoreBundle\Form\Type\TimezoneType;
+use Mautic\LeadBundle\Exception\FieldNotFoundException;
+use Symfony\Component\Form\Extension\Core\Type\DateTimeType;
+use Symfony\Component\Form\Extension\Core\Type\DateType;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\Extension\Core\Type\TimeType;
+use Symfony\Component\Form\Extension\Core\Type\UrlType;
+
+/**
+ * Provides map between Mautic 2 (Symfony 2.8) form aliases and Mautic 3 (Symfony 3.4) FQCN.
+ */
+final class FieldAliasToFqcnMap
+{
+    /**
+     * @format [field alias => field FQCN]
+     */
+    public const MAP = [
```

Comment:
Should this map contain every available type in the New Form Field dropdown list? If so, we're missing Description (`freetext`), Captcha (`captcha`), Free HTML (`freehtml`) and some others. Taken from spot checking the list in the new form form.

Reply:
You may be right. It can be tested by creating such custom field and then loading the contact edit form.

Reply:
These field types are available only for Forms. Not contact/company custom fields. And I tested them, they work.